src/devnet/integrated-devnet-logger.ts

```diff
@@ -0,0 +1,53 @@
+import * as path from "path";
+import * as fs from "fs";
+
+export class IntegratedDevnetLogger {
+
+    constructor(protected stdout?: string, protected stderr?: string) {
+        this.checkFileExists(this.stdout);
+        this.checkFileExists(this.stderr);
+    }
+
+    // Checks if the file exists
+    private async checkFileExists(filePath: string): Promise<void> {
+        if (!filePath || filePath === "STDOUT" || filePath === "STDERR") return;
+        const file = path.resolve(filePath);
+        // Create the file if it doesn't exist
+        const dir = path.dirname(file);
+        if (!fs.existsSync(dir)) {
+            await fs.promises.mkdir(dir, { recursive: true });
+        }
+
+        if (!fs.existsSync(file)) {
+            await fs.promises.writeFile(file, "");
+        }
+    }
+
+    public async logHanlder(logTarget: string, message: string): Promise<void> {
+        if (logTarget === "STDOUT") {
+            console.log(message);
+            return;
+        }
+
+        if (logTarget === "STDERR") {
+            console.error(message);
+            return;
+        }
+
+        // Check if log target is a path to a file
+        if (this.isFile(logTarget)) {
+            // Create the file if it doesn't exist
+            const file = path.resolve(logTarget);
```

Comment:
Why resolve and why resolve here?

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/179/commits/a64c0e0b204d1208f91907181f3838974ef6e8c8